<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainwindow</class>
 <widget class="QMainWindow" name="mainwindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>1000</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>mainwindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QGroupBox" name="AddGeoGroup">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>100</height>
       </size>
      </property>
      <property name="title">
       <string>Add Geometry</string>
      </property>
      <widget class="QPushButton" name="AddBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>71</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>Cube</string>
       </property>
      </widget>
      <widget class="QPushButton" name="AddGrid">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>20</y>
         <width>71</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>Grid</string>
       </property>
      </widget>
      <widget class="QPushButton" name="AddSphere">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>20</y>
         <width>71</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>Sphere</string>
       </property>
      </widget>
      <widget class="QPushButton" name="AddGoal">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>20</y>
         <width>71</width>
         <height>71</height>
        </rect>
       </property>
       <property name="text">
        <string>Goal</string>
       </property>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0" rowspan="2">
       <widget class="QGroupBox" name="ModeGroup">
        <property name="title">
         <string>Tool</string>
        </property>
        <widget class="QPushButton" name="translate">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>70</y>
           <width>41</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>T</string>
         </property>
        </widget>
        <widget class="QPushButton" name="rotate">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>120</y>
           <width>41</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>R</string>
         </property>
        </widget>
        <widget class="QPushButton" name="scale">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>170</y>
           <width>41</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>S</string>
         </property>
        </widget>
        <widget class="QPushButton" name="select">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>41</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string>Arrow</string>
         </property>
        </widget>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QGroupBox" name="SimGroup">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>80</height>
         </size>
        </property>
        <property name="title">
         <string>Simulation</string>
        </property>
        <widget class="QWidget" name="horizontalLayoutWidget">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>331</width>
           <height>61</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QPushButton" name="Run">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Run</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="Stop">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Stop</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="0" column="1" rowspan="2">
       <widget class="SceneView" name="viewer">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QTabWidget" name="tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>350</width>
          <height>0</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="attribute">
         <attribute name="title">
          <string>Tab 1</string>
         </attribute>
         <widget class="QGroupBox" name="groupBox">
          <property name="geometry">
           <rect>
            <x>9</x>
            <y>9</y>
            <width>331</width>
            <height>171</height>
           </rect>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Transform</string>
          </property>
          <widget class="QWidget" name="gridLayoutWidget">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>311</width>
             <height>141</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout">
            <item row="2" column="2">
             <widget class="QLineEdit" name="scale_y">
              <property name="text">
               <string>1.0</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QLineEdit" name="pos_z">
              <property name="text">
               <string>0.0</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="rot_x">
              <property name="text">
               <string>0.0</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="2" column="3">
             <widget class="QLineEdit" name="scale_z">
              <property name="text">
               <string>1.0</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLineEdit" name="rot_y">
              <property name="text">
               <string>0.0</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QLineEdit" name="rot_z">
              <property name="text">
               <string>0.0</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_trans">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>80</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Translate</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_scale">
              <property name="text">
               <string>Scale</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLineEdit" name="scale_x">
              <property name="text">
               <string>1.0</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_rot">
              <property name="text">
               <string>Rotate</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLineEdit" name="pos_y">
              <property name="text">
               <string>0.0</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="pos_x">
              <property name="text">
               <string>0.0</string>
              </property>
              <property name="maxLength">
               <number>5</number>
              </property>
              <property name="cursorPosition">
               <number>3</number>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>190</y>
            <width>331</width>
            <height>181</height>
           </rect>
          </property>
          <property name="title">
           <string>Physical Property</string>
          </property>
          <widget class="QWidget" name="gridLayoutWidget_2">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>311</width>
             <height>151</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label_density">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="MinimumExpanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>80</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Density</string>
              </property>
             </widget>
            </item>
            <item row="3" column="2">
             <widget class="QLineEdit" name="gravity_y">
              <property name="text">
               <string>-9.8</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="4">
             <widget class="QCheckBox" name="check_static">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="MinimumExpanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Static</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_gravity">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Gravity</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="gravity_x">
              <property name="text">
               <string>0.0</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="density">
              <property name="text">
               <string>1</string>
              </property>
             </widget>
            </item>
            <item row="3" column="3">
             <widget class="QLineEdit" name="gravity_z">
              <property name="text">
               <string>0.0</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </widget>
        <widget class="QWidget" name="bullet">
         <attribute name="title">
          <string>Tab 2</string>
         </attribute>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SceneView</class>
   <extends>QOpenGLWidget</extends>
   <header>sceneview.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>pos_x</tabstop>
  <tabstop>pos_y</tabstop>
  <tabstop>pos_z</tabstop>
  <tabstop>rot_x</tabstop>
  <tabstop>rot_y</tabstop>
  <tabstop>rot_z</tabstop>
  <tabstop>scale_x</tabstop>
  <tabstop>scale_y</tabstop>
  <tabstop>scale_z</tabstop>
  <tabstop>density</tabstop>
  <tabstop>check_static</tabstop>
  <tabstop>gravity_x</tabstop>
  <tabstop>gravity_y</tabstop>
  <tabstop>gravity_z</tabstop>
  <tabstop>Run</tabstop>
  <tabstop>Stop</tabstop>
  <tabstop>select</tabstop>
  <tabstop>translate</tabstop>
  <tabstop>rotate</tabstop>
  <tabstop>scale</tabstop>
  <tabstop>AddBox</tabstop>
  <tabstop>AddGrid</tabstop>
  <tabstop>AddSphere</tabstop>
  <tabstop>AddGoal</tabstop>
  <tabstop>tabWidget</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
